debezium-core/src/main/java/io/debezium/transforms/outbox/JsonSchemaData.java

```diff
@@ -127,6 +126,11 @@ private Schema mergeSchema(Schema left, Schema right) {
             if (oldField == null) {
                 fields.put(field.name(), field);
             }
+             else if (oldField.schema().type() == Schema.Type.ARRAY && field.schema().type() == Schema.Type.ARRAY) {
+                 Schema mergedValueSchema = mergeSchema(oldField.schema().valueSchema(), field.schema().valueSchema());
+                 SchemaBuilder arrayBuilder = SchemaUtil.copySchemaBasics(field.schema(), SchemaBuilder.array(mergedValueSchema));
+                 fields.put(field.name(), new Field(field.name(), field.index(), arrayBuilder.build()));
+             }
```

Comment:
Hi @ruthbenjamin, what about situations where the JSON objects in the array not only have missing fields but what if the field types differ, which I don't believe this code change handles:

```
{
  "example": [{
    "code": "300",
    "description": "a description"
  }, {
    "code": "200",
    "description": true
  }]
}
```
In this case, AFAICT the schema is generated solely based on the structure of the first element, which means that the "description" field in the second argument isn't a `boolean` as defined by the JSON.

In my humble opinion, I would expect if the above is the input JSON, I would expect an exception to be thrown given that the nested object has the same field with differing value types.

Reply:
@Naros, I have one question as I am working on Mongo Unwrap SMT: if the JSON is as such 
```
{
  "example": [{
    "code": "300",
    "description": "a description"
  }, {
    "code": "200",
    "description": null
  }]
}
```

Then, as per Mongo bson types, one description would be `BsonString`, and the other would be `BsonNull`, both are different types but would have the same schema type of `String`. In that case, it wouldn't throw an error, right? However, in the above case you mentioned, it would throw the error as the schema types are different too.

Reply:
That's correct @ani-sha